src/inter-aut.cc

```diff
@@ -0,0 +1,410 @@
+    }
+
+    /**
+     * The wrapping function for parsing one section of input to InterAutomaton.
+     * It parses basic information about type of automaton and naming of its component.
+     * Then it parses initial and final formula and finally it creates graphs for transition formula.
+     * @param section A section of input MATA format
+     * @return Parsed InterAutomata representing an automaton from input.
+     */
+    Mata::InterAutomaton mf_to_aut(const Mata::Parser::ParsedSection &section)
+    {
+        Mata::InterAutomaton aut;
+
+        if (section.type.find("NFA") != std::string::npos) {
+            aut.automaton_type = Mata::InterAutomaton::AutomatonType::NFA;
+        } else if (section.type.find("AFA") != std::string::npos) {
+            aut.automaton_type = Mata::InterAutomaton::AutomatonType::AFA;
+        }
+        aut.alphabet_type = get_alphabet_type(section.type);
+
+        for (const auto& keypair : section.dict) {
+            const std::string& key = keypair.first;
+            if (key.find("Alphabet") != std::string::npos) {
+                aut.symbol_naming = get_naming_type(key);
+                if (aut.symbols_enumerated())
+                    aut.symbols_names.insert(
+                            aut.symbols_names.end(), keypair.second.begin(), keypair.second.end());
+            } else if (key.find("States") != std::string::npos) {
+                aut.state_naming = get_naming_type(key);
+                if (aut.states_enumerated())
+                    aut.states_names.insert(
+                            aut.states_names.end(), keypair.second.begin(), keypair.second.end());
+            } else if (key.find("Nodes") != std::string::npos) {
+                aut.node_naming = get_naming_type(key);
+                if (aut.nodes_enumerated())
+                    aut.nodes_names.insert(
+                            aut.nodes_names.end(), keypair.second.begin(), keypair.second.end());
+            }
+        }
+
+        // Once we know what states and nodes are we can parser initial and final formula
+        for (const auto& keypair : section.dict) {
+            const std::string &key = keypair.first;
+
+            if (key.find("Initial") != std::string::npos) {
+                auto postfix = infix2postfix(aut, keypair.second);
+                if (aut.is_nfa() && aut.alphabet_type == Mata::InterAutomaton::EXPLICIT)
+                    postfix = add_operators_implicitly(aut, postfix);
+                aut.initial_formula = postfix2graph(postfix);
+            } else if (key.find("Final") != std::string::npos) {
+                auto postfix = infix2postfix(aut, keypair.second);
+                if (aut.is_nfa() && aut.alphabet_type == Mata::InterAutomaton::EXPLICIT)
+                    postfix = add_operators_implicitly(aut, postfix);
+                aut.final_formula = postfix2graph(postfix);
+            }
+        }
+
+        if (!(!(!(aut.node_naming == Mata::InterAutomaton::Naming::AUTO &&
```

Comment:
The fuck is this monstrosity!

Reply:
I have stopped here for a while as well. Could we at least get a comment explaining this beauty? If I parsed the condition correctly, it should do what it is supposed to do, but… It is not readable at all.

Reply:
it says that at most one of naming types could be automatic.

Reply:
It says, but in the exception message. When the code changes and only this if statement remains here, calling some function in its body, it will take quite some time to figure out the meaning of the statement.

Reply:
any suggestion how to improve it?

Reply:
The code is fine, but I would add a comment explaining the condition, or extract the condition to a function with descriptive name etc.

Reply:
Yes, that would definitely help. My main issue was with the `!(!(!(!!(!(!(!(!(!(!(!(!` madness. I think this test might be candidate for extraction to individual function like `has_atmost_one_auto`

Reply:
extracted to a function.